pkg/rpcclient/wsclient.go

```diff
@@ -283,7 +283,33 @@ func (r *naiveReceiver) EventID() neorpc.EventID {
 
 // Filter implements neorpc.Comparator interface.
 func (r *naiveReceiver) Filter() interface{} {
-	return r.filter
+	if r.filter == nil {
+		return nil
+	}
+	switch f := r.filter.(type) {
+	case *neorpc.BlockFilter:
+		if f == (*neorpc.BlockFilter)(nil) {
```

Comment:
Do you need a type cast here? `f` already has an appropriate type.

Reply:
This was added due to the way how `r.filter` was initialized (https://github.com/nspcc-dev/neo-go/pull/2836#pullrequestreview-1208375867), now this code is removed.